packages/react-core/src/components/DualListSelector/DualListSelector.tsx

```diff
@@ -173,6 +206,39 @@ export class DualListSelector extends React.Component<DualListSelectorProps, Dua
     });
   };
 
+  addTreeSelected = () => {
+    this.setState(prevState => {
+      // Remove selected available nodes from current available nodes
+      const newAvailable = prevState.availableOptions
+        .map(opt => Object.assign({}, opt))
```

Comment:
It's making a copy to prevent array mutation.

Reply:
array filter is already non-mutating and creates a new array. I'd go through and remove unnecessary usages of the map

Reply:
btw, I often reference https://lorenstewart.me/2017/01/22/javascript-array-methods-mutating-vs-non-mutating/

Reply:
`filter` doesn't mutate, but I think it's because of the recursion that it's needed. I tried removing some of the `.map` but then I start losing options when it filters/checks/selects.

Reply:
@kmcfaul Can you use `Array.from(prevState.availableOptions)` instead? It's a little more explicit about the copy.